### PR TITLE
feat: adding proxy server url for amplitude

### PR DESCRIPTION
**Pull request by shrouti1507**
## PR Description

Adding proxy server url config for amplitude. 
ref : https://www.docs.developers.amplitude.com/analytics/domain-proxy/

## Linear task (optional)
resolves INT-1016
[Linear task link](https://linear.app/rudderstack/issue/INT-1016/amplitude-in-device-mode-proxy-the-events-delivery-via-customer)

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-integrations/src/integrations/Amplitude/utils.js [hunk 1]
```diff
@@ -50,4 +50,18 @@ const getFieldsToUnset = integrations => {
...
-export { getTraitsToSetOnce, getTraitsToIncrement, getDestinationOptions, getFieldsToUnset };
+
+/**
+ * Formats the given URL by adding the "https://" prefix if it doesn't already have it.
+ * 
+ * @param {string} url - The URL to be formatted.
+ * @returns {string} - The formatted URL.
+ */
+function formatUrl(url) {
+  if (url.startsWith('http://') || url.startsWith('https://')) {
```
Comment (koladilip) [inline packages/analytics-js-integrations/src/integrations/Amplitude/utils.js:61]: should we even allow http:// (unsecured) endpoint?

Comment (shrouti1507) [reply packages/analytics-js-integrations/src/integrations/Amplitude/utils.js:61]: yes you are right. Removing scope for http

**Review by koladilip — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/Amplitude/utils.js (hunk 1 above)
Comment (yashasvibajpai) [reply packages/analytics-js-integrations/src/integrations/Amplitude/utils.js]: Is it not possible by the customer to not set up a secure tunnel?

**Review by yashasvibajpai — APPROVED** [review]

**Comment by yashasvibajpai** [comment]
lgtm..

File: packages/analytics-js-integrations/__tests__/integrations/Amplitude/browser.test.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import Amplitude from '../../../src/integrations/Amplitude/browser';
 
 const amplitudeConfig = {
   apiKey: 'abcde',
+  proxyServerUrl : 'http://localhost:3000';
```
Comment (koladilip) [inline packages/analytics-js-integrations/__tests__/integrations/Amplitude/browser.test.js:7]: this is http://, expected?

**Review by MoumitaM — DISMISSED** [review]

**Review by yashasvibajpai — DISMISSED** [review]

File: packages/analytics-js-integrations/__tests__/integrations/Amplitude/browser.test.js (hunk 2 above)
Comment (shrouti1507) [reply packages/analytics-js-integrations/__tests__/integrations/Amplitude/browser.test.js:7]: this is just a test case. No, it is not expected. We are using console.error for the user to notify them use of insecured proxy URL.
Screenshot 2024-01-19 at 1 01 54 PM

Comment (shrouti1507) [reply packages/analytics-js-integrations/__tests__/integrations/Amplitude/browser.test.js:7]: While the UI configuration is :
Screenshot 2024-01-19 at 1 04 19 PM

Comment (shrouti1507) [reply packages/analytics-js-integrations/__tests__/integrations/Amplitude/browser.test.js]: edited

**Review by yashasvibajpai — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**